web-ui/src/main/java/org/archcnl/ui/input/ListEntryLayout.java

```diff
@@ -0,0 +1,35 @@
+package org.archcnl.ui.input;
+
+import com.vaadin.flow.component.button.Button;
+import com.vaadin.flow.component.html.Span;
+import com.vaadin.flow.component.orderedlayout.HorizontalLayout;
+
+public class ListEntryLayout<T> extends HorizontalLayout {
+
+    private static final long serialVersionUID = 2L;
+
+    private ListEntry<T> entry;
+
+    private final String EDIT_TEXT = "Edit";
+    private final String DELETE_TEXT = "Delete";
+    private Button editButton;
+    private Button deleteButton;
+
+    public ListEntryLayout(ListEntry<T> entry) {
+        this.entry = entry;
+
+        Span text = new Span(entry.toString());
+        text.setWidth("100%");
+        addAndExpand(text);
+        if (entry.isLeaf()) {
+            editButton = new Button(EDIT_TEXT);
+            deleteButton = new Button(DELETE_TEXT);
```

Comment:
The delete and edit buttons must only be shown for CustomConcepts and CustomRelations as they are the only ones that have an attached mapping and pre-defined relations are not allowed to be deleted.

Reply:
I added a isAlterable() method that for now only returns true, but in the future could be altered to return false for the default objects. The implementation for that seems easier though after the discussion of how we handle/implement the hierarchy.

Reply:
Let's have that discussion tomorrow